src/fusion_power_plant.cc

```diff
@@ -10,14 +10,87 @@ std::string FusionPowerPlant::str() {
   return Facility::str();
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void FusionPowerPlant::EnterNotify() {
+  //Pseudocode implementation of EnterNotify()
+  cyclus::Facility::EnterNotify();
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  //Truncated for readability. It goes on Tritium Storage and gets started, though.
+  fuel_startup_policy.Init(&tritium_storage).Set(fuel_incommod).Start();
+  
+  //Truncated for readability
+  blanket_fill_policy.Init(&blanket).Set(blanket_incommod).Start();
+
+  //Tritium Buy Policy Section:
+  if (refuel_mode == "schedule") {
+    IntDistribution::Ptr active_dist =
+        FixedIntDist::Ptr(new FixedIntDist(1));
+    IntDistribution::Ptr dormant_dist =
+        FixedIntDist::Ptr(new FixedIntDist(buy_frequency - 1));
+    DoubleDistribution::Ptr size_dist =
+        FixedDoubleDist::Ptr(new FixedDoubleDist(1));
+    //Do not start the policy yet.
+    fuel_refill_policy
+        .Init(&tritium_storage, &fuel_tracker,
+              buy_quantity, active_dist, dormant_dist, size_dis)
+        .Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    //otherwise we just do it normally, but still don't start it.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else {
+    //This can be more sophisticated later.
+    throw cyclus::KeyError("Refill mode not recognized!");
+  }
+
+  tritium_sell_policy.Init(&tritium_excess).Set(fuel_incommod).Start();
+  helium_sell_policy.Init(&helium_storage).Set(he3).Start();
+
+}
+
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tick() {
-  std::cout<<"Hello"<<std::endl;
+  //pseudocode implementation of Tick():
+  
+  if (CheckOperatingConditions()) {
+    fuel_startup_policy.Stop();
+    fuel_refill_policy.Start();
+    SequesterTritium();
+    OperateReactor();
+  } else {
+    //Some way of leaving a record of what is going wrong is helpful info I think
+    Record(Error);
+  }
+
+ 
+  DecayInventories();
+  ExtractHelium();
+  MoveExcessTritiumToSellBuffer();
+
+  //This maybe belongs in its own function?
+  if (BlanketCycleTime()) {
+    if (blanket.quantity() >= blanket_turnover) {
+      blanket_excess.Push(blanket.Pop(blanket_turnover));
```

Comment:
When do we refill the blanket?

Reply:
right - you buy directly into the blanket - there is no blanket inventory before things go into the core?

Reply:
I think it makes sense to have an inventory of fresh blanket material that is not just automatically in the core and is being inserted and removed as appropriate

Reply:
So the way it has been working is that in the `OperateReactor()` function, we call some other function called `BreedTritium()`, which pulls the ENTIRE blanket out of that `blanket` resbuf, transmutes some of it into Tritium and He4 (the blanket is required to be Enriched Lithium for this version of the model, but we had plans to change that later on) with the amount being determined by what the user sets as the TBR for the facility, and then we go in and extract the tritium from that composition and then put everything else back into `blanket`. 

The reason I modeled it this way was that I was under the impression that there was just some big tank of liquid enriched lithium and effectively a big swirly pipe that ran through the "blanket" region of the core (maybe that's the wrong word but you hopefully know what I'm talking about), and that the liquid in the tank was being pushed through the pipe fast enough that over the course of a month (standard cyclus timestep) all or most of the liquid would go through the pipe, see some neutrons, and head back into the tank. Then, every once in a while you scoop some "depleted" blanket material out of the tank and pour some "fresh" blanket material in. That may not be the case, though.  

What I hear you saying is that we should have a separate resbuf for fresh blanket material storage which would act as a buffer against limited supply or something (?), and then another resbuf which is the "active" blanket and which has the same role as the current `blanket` resbuf has now but which interacts with the reserve, "fresh" blanket? 

Happy to discuss this more during tomorrow's meeting (or we can push it to Friday if the Fusion crowd is something we want to pull into the conversation) if you think that would be helpful. Hopefully this sort of explains what i was thinking, though.